modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/rebalance/DistributionZoneRebalanceEngine.java

```diff
@@ -221,11 +228,27 @@ private WatchListener createDistributionZonesDataNodesListener() {
                 return nullCompletedFuture();
             }
 
-            Set<String> filteredDataNodes = filterDataNodes(
-                    dataNodes,
-                    zoneDescriptor,
-                    distributionZoneManager.nodesAttributes()
-            );
+            Map<UUID, NodeWithAttributes> nodesAttributes = distributionZoneManager.nodesAttributes();
+
+            Set<String> filteredDataNodes = filterDataNodes(dataNodes, zoneDescriptor, nodesAttributes);
+
+            if (LOG.isInfoEnabled()) {
+                List<NodeWithAttributes> filteredOutNodes = dataNodes.stream()
+                        .filter(node -> !filteredDataNodes.contains(node.nodeName()))
+                        .map(node -> nodesAttributes.get(node.nodeId()))
+                        .collect(toList());
+
+                if (!filteredOutNodes.isEmpty()) {
+                    LOG.info(
+                            "Some data nodes were filtered out because they don't match zone's attributes: "
```

Comment:
What do you think about adding to the log info about the whole data nodes set? Will it be more informative or redundant?  Like "Some data nodes were filtered out from data nodes because they don't match zone's attributes: dataNodes={}....